db/migrate/20230126133557_convert_date_to_other_date.rb

```diff
@@ -0,0 +1,9 @@
+class ConvertDateToOtherDate < ActiveRecord::Migration[7.0]
+  def up
+    Page.where(answer_type: "date", answer_settings: nil).update_all(answer_settings: { input_type: "other_date" })
+  end
+
+  def down
+    Page.where(answer_type: "date", answer_settings: { input_type: "other_date" }).update_all(answer_settings: nil)
```

Comment:
Is there any chance that there would have already been pages with answer type 'date' that already had input_type of "other_date" set via forms-admin before the `up` migration was run, and thus would be incorrectly updated by this rollback `down` migration? Might not be important just curious if the `down` will only ever reverse the `up` or if could update additional entries not originally modified by the `up`? Even if that is the case, I'm not sure how important that would be but thought worth asking :)

Reply:
This is something we discussed - we kind of concluded that there isn't really a way to revert certain migrations without storing historical data for them. This either means we explicitly build them without any reverse gear, or we try and get them to revert as much as possible so that anything that could have been a result of the is moved back. 

I don't think in this case it matters, because flipping this migration back and forth will simply bring everything in line - everything that's up to date already will be unchanged, and moving it back and forth will still retain the same detail.